tests/test_sami_conversion.py

```diff
@@ -1,175 +1,112 @@
-
-        self.assert_dfxp_equals(
-            sample_dfxp_from_sami_with_bad_span_align,
-            results
-        )
-
-    def test_sami_to_dfxp_xml_output(self, sample_sami_syntax_error):
-        captions = SAMIReader().read(sample_sami_syntax_error)
-        results = DFXPWriter(relativize=False,
-                             fit_to_screen=False).write(captions)
-
-        assert isinstance(results, str)
-        assert 'xmlns="http://www.w3.org/ns/ttml"' in results
-        assert 'xmlns:tts="http://www.w3.org/ns/ttml#styling"' in results
-
-
-class TestSAMItoWebVTT(WebVTTTestingMixIn):
-    def test_sami_to_webvtt_conversion(
-            self, sample_webvtt_from_sami, sample_sami):
-        caption_set = SAMIReader().read(sample_sami)
-        results = WebVTTWriter(
-            video_width=640, video_height=360).write(caption_set)
-
-        assert isinstance(results, str)
-        self.assert_webvtt_equals(sample_webvtt_from_sami, results)
-
-    def test_sami_with_style_tags_to_webvtt_conversion(
-            self, sample_webvtt_from_sami_with_style,
-            sample_sami_with_style_tags):
-        caption_set = SAMIReader().read(sample_sami_with_style_tags)
-        results = WebVTTWriter(
-            video_width=640, video_height=360).write(caption_set)
-
-        assert isinstance(results, str)
-        self.assert_webvtt_equals(sample_webvtt_from_sami_with_style, results)
-
-    def test_sami_with_css_inline_style_to_webvtt_conversion(
-            self, sample_webvtt_from_sami_with_style,
-            sample_sami_with_css_inline_style):
-        caption_set = SAMIReader().read(sample_sami_with_css_inline_style)
-        results = WebVTTWriter(
-            video_width=640, video_height=360).write(caption_set)
-
-        assert isinstance(results, str)
-        self.assert_webvtt_equals(sample_webvtt_from_sami_with_style, results)
-
-    def test_sami_with_css_id_style_to_webvtt_conversion(
-            self, sample_webvtt_from_sami_with_id_style,
-            sample_sami_with_css_id_style):
-        caption_set = SAMIReader().read(sample_sami_with_css_id_style)
-        results = WebVTTWriter(
-            video_width=640, video_height=360).write(caption_set)
-
-        assert isinstance(results, str)
-        self.assert_webvtt_equals(sample_webvtt_from_sami_with_id_style,
-                                  results)
-
-
-class TestSAMItoMicroDVD(MicroDVDTestingMixIn):
-    def test_sami_to_micro_dvd_conversion(self, sample_microdvd_2, sample_sami):
-        caption_set = SAMIReader().read(sample_sami)
-        results = MicroDVDWriter().write(caption_set)
-
-        assert isinstance(results, str)
-        self.assert_microdvd_equals(sample_microdvd_2, results)
+        self.assert_sami_captions_equal(sample_sami_partial_margins_relativized,
+                                        result)
 
 
 class TestSAMIWithMissingLanguage(SAMITestingMixIn):
     def test_sami_to_sami_conversion(self, sample_sami_with_lang,
```

Comment:
Wouldn't this be a good fit for the TestSAMItoSAMI class, only with different test name (test_missing_language_conversion)?

Reply:
You're right, I must have missed this while refactoring. Thanks for flagging it!
Updated in d4003ee.